model/headers/headers.go

```diff
@@ -135,7 +135,7 @@ func NewHeader(reader io.Reader, readerPrivateKey [chacha20poly1305.KeySize]byte
 	for i := uint32(0); i < header.HeaderPacketCount; i++ {
 		headerPacket, err := NewHeaderPacket(reader, readerPrivateKey)
 		if err != nil {
-			return nil, err
+			continue
```

Comment:
Normally you don't want to continue on error

Reply:
actually that is documented behaviour, because that is how we manage to decrypt for multiple recipients :)

Reply:
Still doesn't make sense.
If `NewHeaderPacket` returns an error, `headerPacket` will  be nil and you end up appending nil to `header.HeaderPackets`.
And after that you bail only if no header packets has been found.

Reply:
ok, I will revisit the logic to see what can be done there.

Reply:
simplest solution is to ignore the error in this case until we figure out where exactly we should bail in this scenario.

Reply:
one

Reply:
i redid the logic a bit and I think i identified where it happens. I added a custom error for it to check it.

Reply:
if we exit early it seems some tests are broken, and I think we can handle those as part of another PR refactoring the code.